### ISSUE TITLE
Build failure: expected to find jar file but didn't

### ISSUE THREAD

**Issue by ilevd** [thheller/shadow-cljs#1204]
When there is '+' sign in a dependency version (e.g., `[com.github.strojure/parsesso "1.2.2+295"]`)  then build failure occurs:
```
[:app] Build failure:
expected to find jar file but didn't
{:jar-path "/home/igor/.m2/repository/com/github/strojure/parsesso/1.2.2 295/parsesso-1.2.2 295.jar"}
ExceptionInfo: expected to find jar file but didn't
        shadow.build.classpath/get-jar-info* (classpath.clj:354)
        shadow.build.classpath/get-jar-info* (classpath.clj:351)
        clojure.core/apply (core.clj:667)
        clojure.core/memoize/fn--6946 (core.clj:6388)
        shadow.build.classpath/get-jar-info-for-url (classpath.clj:372)
        shadow.build.classpath/get-jar-info-for-url (classpath.clj:364)
        shadow.build.classpath/make-jar-resource (classpath.clj:376)
        shadow.build.classpath/make-jar-resource (classpath.clj:374)
        shadow.build.classpath/find-resource-by-name (classpath.clj:1067)
        shadow.build.classpath/find-resource-by-name (classpath.clj:1055)
        shadow.build.classpath/find-resource-for-provide (classpath.clj:1080)
        shadow.build.classpath/find-resource-for-provide (classpath.clj:1074)
        shadow.build.resolve/find-resource-for-symbol* (resolve.clj:552)
        shadow.build.resolve/find-resource-for-symbol* (resolve.clj:542)
        shadow.build.resolve/find-resource-for-symbol (resolve.clj:597)
        shadow.build.resolve/find-resource-for-symbol (resolve.clj:591)
        shadow.build.resolve/resolve-symbol-require (resolve.clj:651)
        shadow.build.resolve/resolve-symbol-require (resolve.clj:647)
        shadow.build.resolve/resolve-require (resolve.clj:697)
        shadow.build.resolve/resolve-require (resolve.clj:693)
...
```

**Comment by thheller** [comment]
Who the heck puts `+` into a version String ...

**Comment by thheller** [comment]
Sorry this lingered so long. Should be fixed as of `2.28.16`.
